lib/iris/fileformats/netcdf.py

```diff
@@ -1839,10 +1904,12 @@ def _create_cf_data_variable(self, cube, dimension_names, local_keys=None,
         if (not cube.has_lazy_data()
                 or self._dataset.file_format in ('NETCDF3_CLASSIC',
                                                  'NETCDF3_64BIT')):
-            # Determine whether there is a cube MDI value.
-            fill_value = None
-            if isinstance(cube.data, ma.core.MaskedArray):
-                fill_value = cube.data.fill_value
+
+            if not packing:
+                # Determine whether there is a cube MDI value.
+                fill_value = None
+                if isinstance(cube.data, ma.core.MaskedArray):
```

Comment:
@jswanljung You've already cached this in the `masked` local variable, calculated earlier.